### PR TITLE
Auto update the top command

### PR DESCRIPTION
**Pull request by okothkongo**
This give the `top function` to mimic the top command by auto updating the
current running processes on the project.Most of the existing
functionalities have been moved to `Toolshed.TopServer` module.The
module(Toolshed.TopServer) has `GenServer` which ensure that the top
after every `1_000 milliseconds` the processes are refreshed.Pressing
the `enter` exits the command

### CODE REVIEW

**Comment by fhunleth** [comment]
Thanks! I'll take a look this weekend.

**Review by mobileoverlord — APPROVED** [review]

**Review by fhunleth — COMMENTED** [review]
@okothkongo Let's move files around a little for organization. Like this:

```
lib/toolshed/top.ex  <-- main entrance to the top utility. No changes except to adjust module names
lib/toolshed/top/server.ex <-- the GenServer responsible running the timer, getting process info, erasing the previous report and writing the new report to the terminal
lib/toolshed/top/report.ex <-- pure functional code for creating the text that goes on the terminal
```

The goal is to make `report.ex` testable. The other files will be tricky to test, but they'll be small.

Since `top` currently doesn't have unit tests, let's not worry about creating them for this PR. Let's just get things organized so that it's possible.

File: lib/toolshed/top_server.ex [hunk 1]
```diff
@@ -0,0 +1,252 @@
+defmodule Toolshed.TopServer do
+  @moduledoc """
+  Find the top processes and refresh them after every 1000 milliseconds
+  """
```
Comment (fhunleth) [inline lib/toolshed/top_server.ex:4]:
```suggestion
  @moduledoc false
```
This module isn't meant to be used by Toolshed users, so turn off the module docs so they don't see it in.

File: lib/toolshed/top.ex [hunk 2]
```diff
@@ -1,218 +1,15 @@
...
   @doc """
-  List the top processes
-
-  Options:
-
-  * `:order` - the sort order for the results (`:reductions`, `:delta_reductions`,
-    `:mailbox`, `:delta_mailbox`, `:total_heap_size`, `:delta_total_heap_size`, `:heap_size`,
-    `:delta_heap_size`, `:stack_size`, `:delta_stack_size`)
-  * `:n`     - the max number of processes to list
+  List the running  processes and auto-update them after every `1_000` milliseconds.Press enter to exit.
```
Comment (fhunleth) [inline lib/toolshed/top.ex:6]:
```suggestion
  Interactively show the top Elixir processes
```

**Review by fhunleth — APPROVED** [review]

**Pull request merged**